lib/include/openamp/virtio.h

```diff
@@ -468,13 +468,15 @@ static inline int virtio_negotiate_features(struct virtio_device *vdev,
 					    uint32_t features,
 					    uint32_t *final_features)
 {
-	if (!vdev || !final_features)
+	if (!vdev)
 		return -EINVAL;
 
 	if (!vdev->func || !vdev->func->negotiate_features)
 		return -ENXIO;
 
-	*final_features = vdev->func->negotiate_features(vdev, features);
+	vdev->features = vdev->func->negotiate_features(vdev, features);
+	if (final_features)
+		*final_features = vdev->features;
```

Comment:
Here you try 2 address 2 things 
- make `final_features `optional
- update `vdev->features`

It look like this PR should contain 4 patches

- patch 1:  update `virtio_dispatch` struct  comments to specify that `negotiate_features` return negociated feature in case of success , negative value in case of error

- patch 2:  fix rproc_virtio_negotiate_features that retun 0 instead of dfeatures & features

- patch 3: set `vdev->features` in `virtio_negotiate_features `
- patch 4: make `final_features` param optional ( code + parameter comment)

Reply:
> Here you try 2 address 2 things
> 
> * make `final_features `optional
> * update `vdev->features`
> 
> It look like this PR should contain 4 patches
> 
> * patch 1:  update `virtio_dispatch` struct  comments to specify that `negotiate_features` return negociated feature in case of success , negative value in case of error
> * patch 2:  fix rproc_virtio_negotiate_features that retun 0 instead of dfeatures & features
> * patch 3: set `vdev->features` in `virtio_negotiate_features `
> * patch 4: make `final_features` param optional ( code + parameter comment)

Ok,  For the 4 patches there are some issues：
patch 1: negotiate_features in virtio_dispatch return type is uint32_t, so can't negative value in case of error.
patch 3 and patch 4: can we  merge to one directly like the commit?

Reply:
Ok, done.